src/utils/utils.ts

```diff
@@ -181,3 +186,29 @@ export const checkColumnsCombinations = ({ columns, combinations }: ICheckColumn
     }
   });
 };
+
+type TExtendString = string & { _excludeType: 'q' };
+
+interface IExtendQParam {
+  id: string;
+  value: string;
+  operator: IQParamOperators;
+}
+
+interface IExtendParams {
+  qParams?: IExtendQParam[];
+  [key: TExtendString]: string;
+}
+
```

Comment:
I would expect that I should be able to use for example:
```tsx
const newReqestConfig = extendRequestConfig(requestConfig, { timeout: 123 });
```

but it will result in:
```
Argument of type '{ timeout: number; }' is not assignable to parameter of type 'IExtendParams'.
  Object literal may only specify known properties, and 'timeout' does not exist in type 'IExtendParams'.
```